api/app_sph_lms/api/serializer/learning_path_serializer.py

```diff
@@ -0,0 +1,81 @@
+from app_sph_lms.models import (Category, Course, LearningPath,
+                                LearningPathCourse)
+from rest_framework import serializers
+
+
+class LearningPathCourseSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = LearningPathCourse
+        fields = ('course', 'course_order')
+
+
+class LearningPathSerializer(serializers.ModelSerializer):
+    courses = serializers.ListField(
+      child=LearningPathCourseSerializer(),
+      write_only=True
+    )
+    category = serializers.PrimaryKeyRelatedField(
+        queryset=Category.objects.all(),
+        many=True,
+    )
+
+    class Meta:
+        model = LearningPath
+        exclude = ['author']
+
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation['category'] = [
+            {
+                'id': category.id,
+                'name': category.name
+            } for category in instance.category.all()
+        ]
+        return representation
+
```

Comment:
```suggestion
class LearningPathSerializer(serializers.ModelSerializer):
    category = CategorySerializer(many=True, read_only=True)

    class Meta:
        model = LearningPath
        exclude = ['author']

```

I think we are not using courses in learning path list, when it comes to course count, since we have a relation with courses now in model, they can just get it by `learning_path.courses.all().count()`

and also, since we have a relation with categories in model, you can just use existing category serializer to get the categories in the desirable format, it does the trick without any need of `to_representation`

Reply:
I'll try adjusting using the `categories = CategorySerializer(many=True, read_only=True)` 🙇‍♀️